packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -567,6 +570,55 @@ export function authorizationServiceBuilder(
         showUsers: true,
       };
     },
+    async setAdminToClient(
+      {
+        clientId,
+        adminId,
+      }: {
+        clientId: ClientId;
+        adminId: UserId;
+      },
+      { authData, correlationId, logger }: WithLogger<AppContext<UIAuthData>>
+    ): Promise<Client> {
+      assertFeatureFlagEnabled(config, "featureFlagAdminClient");
+
+      logger.info(`Set user ${adminId} in client ${clientId} as admin`);
+
+      await assertUserSelfcareSecurityPrivileges({
+        selfcareId: authData.selfcareId,
+        requesterUserId: authData.userId,
+        consumerId: authData.organizationId,
+        userIdToCheck: adminId,
+        selfcareV2InstitutionClient,
+        correlationId,
+        userRolesToCheck: [userRole.ADMIN_ROLE],
+      });
+
+      const client = await retrieveClient(clientId, readModelService);
+      assertClientIsAPI(client.data);
+      assertOrganizationIsClientConsumer(authData, client.data);
+
+      const oldAdminId = client.data.adminId;
+      if (oldAdminId && oldAdminId === adminId) {
+        throw clientAdminAlreadyAssignedToUser(clientId, adminId);
+      }
+
+      const updatedClient: Client = {
+        ...client.data,
+        adminId,
+      };
+
+      await repository.createEvent(
+        toCreateEventClientAdminSet(
+          adminId,
+          updatedClient,
+          client.metadata.version,
+          correlationId,
+          oldAdminId
+        )
+      );
+      return updatedClient;
```

Comment:
```suggestion
      return { updatedClient, showUsers: true }
```
We should show the users since we already asserted that `OrganizationIsClientConsumer`, I would return this to the router and then use this `showUsers` inside `clientToApiClient`